### DISCUSSION TITLE
Unreliable Zigbee network

### DISCUSSION THREAD

**Discussion by Ra72xx** [Koenkk/zigbee2mqtt#19882] [General]
I switched from Deconz SW/HW to Zigbee2MQTT and SkyConnect a few weeks ago because the Deconz solution is somewhat restricted. Unfortunately, I seem to have traded in for a more unreliable solution. Yes, I know EZSP support is only in beta, yes; I have followed all best-practice hints for Zigbee networks (2m extension, USB 2.0, away from router, etc.).

Symptoms are:

- Devices dropping silently (!) out ot the network - no information about this from Z2M, but devices can at least be re-paired keeping all settings when joining them again.
- Communication dropping randomly - especially Aqara thermostats are prone to this: According to Z2M everything is fine, but obviously data/commands have not been transmitted to the device.
- Very bad recovery from network changes - I can bet that after every power loss or e.g. movement of a switchable outlet in my flat I have go searching for missing devices.
- Devices missing endpoints which should be present according to UI, e.g. I have seemingly  lost the possibility to do ota updates a few days ago 

If you scan github issues, there seem to be quite a few of those issues, seemingly especially starting with version 1.33 and seemingly - but not exclusively! - with ezsp coordinatos. However, these bug reports seem to go unnoticed, which I find a bit unnerving, as this more-or-less silent network degradation is a very bad situation. I can't rely any longer on my fundamental smarthome network.

Has anybody experienced this, too, are there know workarounds, should I maybe switch to ZHA for my SkyConnect device? How to priorize an error like this?

**Comment by oywino** [comment]
I'm not sure if my own problems are in any way related to yours, but after running a stable Z2M network for a very long time, one day I found the Z2M server had crashed and refused to start at all, with the following error message, never before seen here: 
`
MQTT retention requires protocol version 5
`
I have not been able to figure out exactly what this means, and why this error suddenly popped up out of nowhere one day. But I found several other on the forum with the same issue. A common solution seemed to be just to add 
`
version: 5
`
to the mqtt section in _configuration.yaml_. Then the Z2M started again, but has since been very unstable. It is loosing connection to my ZB-Coordinator (a Sonoff Bridge flashed with _Tasmota v13.2_ and _ncp-uart-sw_6.7.8_115200_) resulting in frequent Z2M chrashes (!?) and it then refused to start again with the following error message:
```
Error while starting zigbee-herdsman
Failed to start zigbee
```
If I manually restarted the coordinator, then the Z2M started again, and would run fine but only for a few hours before again chrashing.
I tried updating Z2M, but to no avail. It remained unstable and crashed randomly after running fine for about  2-4 hours. This is my current setup:
```
Zigbee2MQTT version  1.33.2 commit: [9996c93]
Coordinator type EZSP v8
Coordinator revision 6.7.8.0 build 373
Frontend version 0.6.142
Zigbee-herdsman-converters version 15.106.0
Zigbee-herdsman version 0.21.0

```
Like you, my ZB network has gone from being rock solid to completely useless.
It all happened after the `version 5` issue popped up.

**Comment by Ra72xx** [comment]
That seems to be a different problem, as it concerns not the Zigbee network itself but the way Z2M interacts with MQTT. 
The situation could, however, be considered similar as Z2M gets into an problematic situation (in my case: lost connections and devices) and does not inform the user. 
In your case Z2M could even solve the problem by itself by setting the correct parameters (e.g. disabling retention or setting an appropriate MQTT version) in order to get at least basic functionality and then informing the user of this fact.

**Reply by oywino** [reply]
I agree. I have now removed the `version: 5` settng and deleted the retention for the one specific device. Let's see how that works out. I'll report back. 
But to your problem with lost devices. I too had that problem after every crash. It turns out Z2M for some obscure reason wipes the content from the `database.db` file in the config folder (!?)
"Luckily" there is a `database.db.backup` which I copied over the empty `database.db` file, and restarted Z2M and all the missing devices was back.
Why Z2M does something as crazy as wiping it's DB file, is beyond me. Perhaps such behavior should be reported as a bug?

**Comment by Ra72xx** [comment]
Problem seems to be not so uncommon: #19747

**Comment by oywino** [comment]
> Problem seems to be not so uncommon: #19747

No, that's not the same issue. I have not seen my issue with wiped database anywhere.

**Reply by Ra72xx** [reply]
Well, my original post was about unreliable Zigbee network, and that was what I was referring to :-) .

**Reply by oywino** [reply]
Aha, I see  - sorry for misunderstanding.
And I agree - there are many (too many?) with stability problems. What puzzles me is why Z2M chooses to go belly-up for even small network irregularities?

**Comment by Ra72xx** [comment]
Well, it seems that there are even worse bugs than losing communication now and then :-/ ... I don't know if that's a similar problem.
https://github.com/Koenkk/zigbee2mqtt/issues/19988
